missilewars-plugin/src/main/java/de/butzlabben/missilewars/util/geometry/GameArea.java

```diff
@@ -0,0 +1,156 @@
+        
+        this.world = center.getWorld();
+        
+        long x1 = center.getBlockX() + offset;
+        long x2 = center.getBlockX() - offset;
+        long z1 = center.getBlockZ() + offset;
+        long z2 = center.getBlockZ() - offset;
+
+        long y1 = center.getBlockY();
+        long y2 = y1 + height;
+
+        this.position1 = new Location(center.getWorld(), x1, y1, z1);
+        this.position2 = new Location(center.getWorld(), x2, y2, z2);
+
+        initialize();
+    }
+
+    /**
+     * This method creates a new GameArena object.
+     *
+     * The GameArena is a rectangular area. His border (1 block wide) is 
+     * still part of the arena (= inside).
+     *
+     * @param world (World) the target world for the desired area
+     * @param areaConfig (AreaConfiguration) the loaded Area-Configuration from which the data is taken
+     */
+    public GameArea(World world, AreaConfiguration areaConfig) {
+        
+        this.world = world;
+        
+        this.position1 = new Location(world, areaConfig.getMinX(), areaConfig.getMinY(), areaConfig.getMinZ());
+        this.position2 = new Location(world, areaConfig.getMaxX(), areaConfig.getMaxY(), areaConfig.getMaxZ());
+
+        if (position1.equals(position2)) throw new IllegalArgumentException("The selected positions do not differ.");
+        
+        initialize();
+    }
+
+    /**
+     * This method calculates and saves the MIN and MAX positions 
+     * according to the current values. The assigned MIN and MAX 
+     * information can be used to later compare the GameArea more 
+     * easily with current live positions/areas.
+     */
+    private void initialize() {
+
+        if (position1.getBlockX() < position2.getBlockX()) {
+            minX = position1.getBlockX();
+            maxX = position2.getBlockX();
+        } else {
+            maxX = position1.getBlockX();
+            minX = position2.getBlockX();
+        }
+
+        if (position1.getBlockY() < position2.getBlockY()) {
+            minY = position1.getBlockY();
+            maxY = position2.getBlockY();
+        } else {
+            maxY = position1.getBlockY();
+            minY = position2.getBlockY();
+        }
+
+        if (position1.getBlockZ() < position2.getBlockZ()) {
+            minZ = position1.getBlockZ();
+            maxZ = position2.getBlockZ();
+        } else {
+            maxZ = position1.getBlockZ();
+            minZ = position2.getBlockZ();
+        }
+    }
+
+    public AreaConfiguration getAreaConfiguration() {
+        AreaConfiguration newAreaConfig = new AreaConfiguration(position1.getBlockX(), position1.getBlockY(), position1.getBlockZ(), 
+                position2.getBlockX(), position2.getBlockY(), position2.getBlockZ());
+        return newAreaConfig;
+    }
+
+    // currently not used; perhaps useful in the future
+    private enum Direction {
+
+        NULL,
```

Comment:
What is `NULL` exactly? We don't know it? We haven't determined it yet?
Please use more descriptive names/add some JavaDoc.

Reply:
I thought I'd just build a Bais class with default values. Rotations/Directions would be suitable there.